eva-tools/src/main/java/embl/ebi/variation/eva/vcfDump/VariantExporter.java

```diff
@@ -70,23 +89,26 @@
      * If there won't be empty alleles, CellBase is not needed, and the parameter may be null.
      *
      * @param cellbaseClient for empty alleles. nullable.
+     * @param variantDBAdaptor
+     * @param query
      */
-    public VariantExporter(CellBaseClient cellbaseClient) {
+    public VariantExporter(CellBaseClient cellbaseClient, VariantDBAdaptor variantDBAdaptor, QueryOptions query) {
         this.cellbaseClient = cellbaseClient;
+        this.variantDBAdaptor = variantDBAdaptor;
+        this.options = query;
+        this.regionDivider = new RegionDivider(WINDOW_SIZE);
     }
 
     /**
      * Main method of this class. It generates one VCF file per requested study,
      * writing both the header meta-data and the variants in the body.
      *
-     * @param iterator where to get the variants from
+
      * @param outputDir directory to write the output vcf(s) to
      * @param sourceDBAdaptor to retrieve all the VariantSources in any VariantSourceEntry
-     * @param options not implemented yet, use only for studyId and fileId
      * @return list of files written, one per study.
      */
-    public List<String> VcfHtsExport(Iterator<Variant> iterator, String outputDir,
-                                     VariantSourceDBAdaptor sourceDBAdaptor, QueryOptions options) throws IOException {
+    public List<String> VcfHtsExport(String outputDir, VariantSourceDBAdaptor sourceDBAdaptor) throws IOException {
```

Comment:
The public interface should have a name with no references to implementation details (using the HTS library), like `export`.